server/src/modules/users/mappers/auth-code-map.ts

```diff
@@ -0,0 +1,30 @@
+import { UniqueEntityID } from '../../../shared/domain/unique-entity-id'
+import { AuthCodeEntity } from '../../../shared/infra/db/entities/auth-code.entity'
+
+import { AuthCode } from '../domain/entities/auth-code/auth-code'
+import { AuthCodeString } from '../domain/value-objects/auth-code'
+
+export class AuthCodeMap {
+
+  public static toDomain(authCodeEntity: AuthCodeEntity): AuthCode {
+
+    const authCode = AuthCode.create(
+      {
+        clientId: authCodeEntity.clientId,
+        authCode: new AuthCodeString(authCodeEntity.authCode)
+      },
+      new UniqueEntityID(authCodeEntity.id)
+    )
+    if (authCode.isErr()) throw new Error() // TODO: check if we should handle error differently
+      
+    return authCode.value
+  }
+
+  public static async toPersistence(authCode: AuthCode): Promise<AuthCodeEntity> {
+    const authCodeEntity = new AuthCodeEntity()
+    authCodeEntity.clientId = authCode.clientId
+    authCodeEntity.authCode = authCode.authCode.getValue()
```

Comment:
should we change the name of the property to be `authCodeString` instead? This is a little confusing as to what is actually being used as the `authCode` value here.

Reply:
Good catch.